Third party > monthly report tab

christoskaterini opened the pull request:
# NEW|New Monthly Report feature for third parties

Added a comprehensive monthly report functionality for third-party companies that provides detailed invoice analysis across multiple years with filtering capabilities and PDF export.

## Features Added:

### Core Functionality:
- **Monthly Report Tab**: New tab in third-party view (`societe/monthly_report.php`)
- **Multi-year Analysis**: Displays last 3 years of invoice data in tabular format
- **Invoice Type Filtering**: Filter by client invoices, supplier invoices, or both
- **Percentage Change Calculations**: Shows month-over-month and year-over-year percentage changes
- **Total Summaries**: Yearly totals with percentage comparisons

### User Interface:
- **Elegant Table Design**: Clean borders with year separators for easy reading
- **Responsive Layout**: Table adapts to different screen sizes
- **Color-coded Changes**: Green for positive changes, red for negative changes
- **Professional Styling**: Matches Dolibarr's design standards

### Export & Print:
- **PDF Export**: Clean print-friendly PDF generation with proper formatting
- **Print View**: Optimized print layout with close button functionality
- **Filtered Exports**: PDF respects selected filters and date ranges

### Technical Implementation:
- **Optimized SQL Queries**: Efficient database queries with proper indexing
- **Credit Note Support**: Includes all invoice types (validated, paid, credit notes)
- **Security**: Proper input validation and SQL injection prevention
- **Database Compatibility**: Uses Dolibarr's database abstraction layer

### Files Modified/Added:
- `societe/monthly_report.php` (NEW)
- Tab integration with existing third-party interface

This feature enhances Dolibarr's financial reporting capabilities by providing detailed monthly analysis tools for better business intelligence and customer relationship insights.

christoskaterini committed:
Add files via upload

christoskaterini committed:
Add files via upload

aspangaro commented:
Hello, thanks for your contribution.

Can provide a picture of the new page ?

Why add a file companies.lang for language at the root of the langs folder ? Only add new language key in en_US/compagnies.lang. (en_US is the default language)

Perhaps this page should be placed in the ‘societe’ rather than ‘core’  folder, as it concerns third parties.

How is it intended that this page be accessed?

christoskaterini committed:
Merge branch 'Dolibarr:develop' into third_party_monthly_report

christoskaterini committed:
fix files position

christoskaterini commented:
Hellow @aspangaro and thank you for your comments. 

1. Removed the extra companies.lang file and added the new keys inside en_US/companies.lang.
2. Moved the page into the societe folder.
3. Updated htdocs/core/lib/company.lib.php so the new tab is properly accessible.

That’s what happens when you create a PR at 04:00 in the morning 😅, but everything should now be corrected and aligned with the project structure.

<img width="1920" height="1482" alt="εικόνα" src="https://github.com/user-attachments/assets/5616948c-80c1-4921-9b21-b112f72f5b91" />

christoskaterini committed:
add missing translations

christoskaterini committed:
Merge branch 'Dolibarr:develop' into third_party_monthly_report

christoskaterini committed:
Merge branch 'Dolibarr:develop' into third_party_monthly_report

christoskaterini committed:
fix (codesniffer)

eldy committed:
Merge branch 'develop' into third_party_monthly_report

aspangaro commented:
Thanks for your returns.

Following your changes, there are duplication issues with the language keys:
<img width="556" height="527" alt="image" src="https://github.com/user-attachments/assets/80f368d5-3e55-425f-82e0-c80b87c1786f" />

You need to add 'cashdesk' file language in your $langs->loadLangs() and remove all of the above keys from your companies.lang file

Thanks !

christoskaterini committed:
Merge branch 'Dolibarr:develop' into third_party_monthly_report

christoskaterini committed:
cashdesk.lang update

christoskaterini commented:
Thanks for the feedback.

I’ve removed all the duplicate translations from companies.lang and updated the page to load the cashdesk language file with:

```
$langs->loadLangs(array("companies", "bills", "cashdesk"));
```

This way it now uses the existing translations instead of duplicating them. Everything should be aligned with Dolibarr’s structure now.

christoskaterini committed:
fix

christoskaterini committed:
Merge branch 'Dolibarr:develop' into third_party_monthly_report

christoskaterini committed:
Merge branch 'Dolibarr:develop' into third_party_monthly_report
